File: src/main/scala/im/tox/toktok/app/simple_dialogs/SimpleTextDialogDesign.scala
package im.tox.toktok.app.simple_dialogs

import android.app.{ Activity, Dialog }
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View.OnClickListener
import android.view.Window
import im.tox.toktok.TypedResource._
import im.tox.toktok.{ R, TR }
import org.scaloid.common._

final class SimpleTextDialogDesign(
    activity: Activity,
    title: String,
    color: Int,
    icon: Int,
    value: String,
    clickAction: OnClickListener
) extends Dialog(activity, R.style.DialogSlideAnimation) {

  override def onCreate(savedInstanceState: Bundle): Unit = {
    super.onCreate(savedInstanceState)
    requestWindowFeature(Window.FEATURE_NO_TITLE)
    setContentView(R.layout.simple_input_dialog_design)
    getWindow.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT))

    this.findView(TR.simple_dialog_color).setBackgroundTintList(ColorStateList.valueOf(color))
    this.findView(TR.simple_dialog_img).setImageResource(icon)
    this.findView(TR.simple_dialog_text).setText(title)

    val input = this.findView(TR.simple_dialog_input)
    input.setText(value)

    val cancelButton = this.findView(TR.simple_dialog_cancel)
    cancelButton.onClick {
      dismiss()
    }

    val confirmButton = this.findView(TR.simple_dialog_confirm)

    input.afterTextChanged {
      confirmButton.setEnabled(true)
      confirmButton.setTextColor(activity.getResources.getColor(R.color.simpleDialogTextButton, null))
      confirmButton.setBackgroundTintList(ColorStateList.valueOf(activity.getResources.getColor(R.color.simpleDialogIconButton, null)))
      confirmButton.setOnClickListener(clickAction)
    }
  }
}

File: src/main/java/im/tox/toktok/app/simple_dialogs/SimpleTextDialogDesign.java
package im.tox.toktok.app.simple_dialogs;

import android.app.Activity;
import android.app.Dialog;
import android.content.res.ColorStateList;
import android.content.res.Resources;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import androidx.annotation.NonNull;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import im.tox.toktok.R;
import im.tox.toktok.app.CompatUtil;

public final class SimpleTextDialogDesign extends Dialog {

    @NonNull
    private final Activity activity;
    private final String title;
    private final int color;
    private final int icon;
    private final String value;
    private final View.OnClickListener clickAction;

    public SimpleTextDialogDesign(
            @NonNull Activity activity,
            String title,
            int color,
            int icon,
            String value,
            View.OnClickListener clickAction
    ) {
        super(activity, R.style.DialogSlideAnimation);
        this.activity = activity;
        this.title = title;
        this.color = color;
        this.icon = icon;
        this.value = value;
        this.clickAction = clickAction;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.simple_input_dialog_design);
        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));

        this.findViewById(R.id.simple_dialog_color).setBackgroundTintList(ColorStateList.valueOf(color));
        this.<ImageView>findViewById(R.id.simple_dialog_img).setImageResource(icon);
        this.<TextView>findViewById(R.id.simple_dialog_text).setText(title);

        EditText input = this.findViewById(R.id.simple_dialog_input);
        input.setText(value);

        Button cancelButton = this.findViewById(R.id.simple_dialog_cancel);
        cancelButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dismiss();
            }
        });

        final Button confirmButton = this.findViewById(R.id.simple_dialog_confirm);
        input.addTextChangedListener(new TextWatcher() {
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {
                confirmButton.setEnabled(true);
                final Resources resources = activity.getResources();
                confirmButton.setTextColor(CompatUtil.getColor(resources, R.color.simpleDialogTextButton));
                confirmButton.setBackgroundTintList(ColorStateList.valueOf(CompatUtil.getColor(resources, R.color.simpleDialogIconButton)));
                confirmButton.setOnClickListener(clickAction);
            }

            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }
        });
    }
}
